### ISSUE TITLE
Compiling for 64 bits windows 7

### ISSUE THREAD

**Issue by mhinne** [brian-lau/MatlabStan#14]
Dear all,

I have installed CmdStan using MinGW (64 bits) on Cygwin on 64 bits Windows 7. This worked fine and the Bernoulli example compiled and executed without errors. Now, I tried to use the schools example in Matlab, but this gives the following error(s): 

```
Error using StanModel/compile (line 852)
File not found - *_inst.cpp
File not found - *_def.cpp
File not found - *.stan
File not found - *.hpp
FIND: Parameter format not correct
Assembler messages:
Fatal error: No compiled in support for x86_64
src/cmdstan/main.cpp:1:0: sorry, unimplemented: 64-bit mode not compiled in
src/cmdstan/main.cpp:13:1: fatal error: error closing -: Invalid argument
compilation terminated.
```

Apparently, something goes wrong with the compilation, but as the Bernoulli example worked fine, I'm unsure what went wrong. 

Would anyone have a clue? :-)

Thanks!

**Comment by brian-lau** [comment]
When you get this error, is there a *.stan file in your working directory? Does your working directory have any spaces in the name?

**Comment by mhinne** [comment]
Hmm no to both questions. After running a model, a *.stan file was created in the pwd though (anon_model.stan). Subsequent runs gave a pop-up asking for a new file name.

I'm testing this on two different computers, both Windows 7, 64 bits (two different work places). The second computer gives the following error instead: 

    Error using StanModel/stan_version
    Output argument "ver" (and maybe others) not assigned during call to
    "C:\Users\Max\Documents\MATLAB\Packages\MatlabStan-    2.7.0.0\StanModel.m>StanModel.stan_version".

Not sure if it's related of course, but the hardware and OS of both systems is identical.

**Comment by mhinne** [comment]
Perhaps related (although it wouldn't appear so) is the warning that precedes this error:

    Warning: # of chains > # of cores. 
    > In StanModel>StanModel.set.chains at 470
      In StanModel>StanModel.StanModel at 224
      In stan at 108 

Even though I'm only running 1 chain.

**Comment by brian-lau** [comment]
I'm still confused. Would you be so kind as to try running the model outside of Matlab at the command line? The .stan file and data are here: 

https://github.com/stan-dev/example-models/tree/master/misc/eight_schools

Thanks.

**Comment by mhinne** [comment]
That works like a charm!

**Comment by mhinne** [comment]
So I'm basically getting different errors on both systems (they are far apart so difficult to try both at the same time). System one gives the 'no compiled in support x86_64' error, system two gives the problem with getting the Stan version. 

I'm currently on system two, where I've now tried:

* changing the cmdstan installation to a path without spaces: no changes
* I can get the stanc version via cygwin, which I used to compile cmdstan with. However, when I use 'stanc --version' in the windows command utility, I get the error 'application unable to start correctly'. I assume this is what MatlabStan uses under the hood, and is where the problem arises.

**Comment by brian-lau** [comment]
Thanks for the update. For system two, you are correct, I'm calling down to stanc to get the version. I would have assumed that this would prevent you from running the eight_schools or bernoulli examples from the command line as well?

**Comment by mhinne** [comment]
I'm sorry for the confusion: I have tried the eight-schools and Bernoulli example using cygwin. Here, the cmdstan tool works as intended (and also gives me the correct version). However, Matlab presumably uses the windows Cmd tool (or something similar), in which this goes awry.

I have now found a dirty hack for system two, which is to simply put 
     ver =  '2.7.0';
in the version() function in StanModel.m. Apparently, stanc performs fine otherwise, because now I can run everything from within Matlab.

I'll return to system one on Monday. I'm suspecting I have some libraries messed up.

Thanks a lot for your support by the way - the tool is great when it works! :-)
